File: JavaScript/Medium/two_pointers/prefixSuffixEqual.js
var solution = function(s){
    let right = s.length -1
    let left = 0

    while (left < right && s.charAt(left) === s.charAt(right)){
        let char = s.charAt(left)
        while (left <= right && s.charAt(left) == char){
            left ++
        }
        while (left <= right && s.charAt(right) == char){
            right --
        }
    }

    return 0 ? left > right : right - left + 1
}

File: Python/Medium/two_pointers/prefixSuffixEqual.py
def solution(s):
    left = 0
    right = len(s) - 1

    while left < right and s[left] == s[right]:
        char = s[left]
        while left <= right and s[left] == char:
            left +=1
        while left <= right and s[right] == char:
            right -= 1

    return 0 if left > right else right - left + 1